src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ecm.common.helpers.UtilHelper;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseDetails;
+import uk.gov.hmcts.ecm.common.model.multiples.SubmitMultipleEvent;
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.MultiplesHelper;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.excel.MultipleCasesReadingService;
+
+import java.util.List;
+
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.MULTIPLE_CASE_TYPE;
+
+@RequiredArgsConstructor
+@Service("fixCaseApiService")
+public class FixCaseApiService {
+
+    private final MultipleCasesReadingService multipleCasesReadingService;
+
+    @Value("${ccd_gateway_base_url}")
+    private String ccdGatewayBaseUrl;
+
+    public void checkUpdateMultipleReference(CaseDetails caseDetails, String userToken) {
+        CaseData caseData = caseDetails.getCaseData();
+        if (caseData.getEcmCaseType().equals(MULTIPLE_CASE_TYPE)) {
+            String multipleCaseTypeId = UtilHelper.getBulkCaseTypeId(caseDetails.getCaseTypeId());
+            List<SubmitMultipleEvent> submitMultipleEvents = multipleCasesReadingService.retrieveMultipleCases(
+                    userToken, multipleCaseTypeId, caseData.getMultipleReference());
+            if (!submitMultipleEvents.isEmpty()) {
+                String multipleCaseId = String.valueOf(submitMultipleEvents.get(0).getCaseId());
+                caseData.setMultipleReferenceLinkMarkUp(getLinkMarkUp(multipleCaseId, caseData.getMultipleReference()));
```

Comment:
I think this means you need to change ecm-common. The ES query is using the field name "reference" but you are not passing the CCD reference. Instead you are passing multipleReference (this is a reference generated by ECM). Therefore, you need to change the ES query to filter on that field